peregrine/iqgen/bits/doppler_const.py

```diff
@@ -0,0 +1,196 @@
+    userTime_s : float
+      Observer's time in seconds
+
+    Returns
+    -------
+    float
+      Satellite vehicle's time at which signal has been generated.
+    '''
+
+    tau_s = self.c1 + self.c2 * userTime_s
+    svTime_s = userTime_s - tau_s
+
+    return svTime_s
+
+  def computeBatch(self,
+                   userTime0_s,
+                   n_samples,
+                   amplitude,
+                   planFrequency_hz,
+                   ifFrequency_hz,
+                   message,
+                   code):
+    '''
+    Computes signal samples for the doppler object.
+
+    Parameters
+    ----------
+    userTime0_s : float
+      Observer's time in seconds of the beginning of the interval.
+    n_samples : int
+      Number of samples to generate
+    amplitude : float
+      Signal amplitude.
+    planFrequency_hz : float
+      Central carrier frequency in hertz
+    ifFrequency_hz: float
+      Intermediate frequency in hertz
+    message : object
+      Message object for providing access to symbols
+    code : object
+      PRN code object for providing access to chips
+
+    Returns
+    -------
+    signal : numpy.ndarray(n_samples, dtype=float)
+      Generated samples
+    userTimeX_s : float
+      End of interval time in seconds
+    chipAll_idx : numpy.ndarray(n_samples, dtype=float)
+      Code chip phases for the samples
+    chips : numpy.ndarray(n_samples, dtype=int)
+      Code combined with data
+    '''
+    deltaUserTime_s = float(n_samples) / if_iface.Chip.SAMPLE_RATE_HZ
+    userTimeX_s = userTime0_s + deltaUserTime_s
+    tau0_s = self.c1 + self.c2 * userTime0_s  # Initial time difference
+    tauX_s = self.c1 + self.c2 * userTimeX_s  # Final time difference
+
+    svTime0_s = userTime0_s - tau0_s  # Start time
+    svTimeX_s = userTimeX_s - tauX_s  # End time
+
+    # Compute initial and final signal phases and then phase space
+    doppler_hz = -planFrequency_hz / scipy.constants.c * self.speed_mps
+    phase0_s = scipy.constants.pi * 2. * (ifFrequency_hz + doppler_hz) * userTime0_s  # svTime0_s
+    phaseX_s = scipy.constants.pi * 2. * (ifFrequency_hz + doppler_hz) * userTimeX_s  # svTimeX_s
+    signal = scipy.linspace(phase0_s, phaseX_s, n_samples, endpoint=False)
+
+    # Convert phase to signal value and multiply by amplitude
+    scipy.sin(signal, signal)
+    scipy.multiply(signal, amplitude, signal)
+
+    # PRN and data index computation
+    chip0_idx = svTime0_s * 1023000.
+    chipX_idx = svTimeX_s * 1023000.
+    chipAll_idx = scipy.linspace(chip0_idx, chipX_idx, n_samples, endpoint=False)
+
+    def dataChip(idx):
+      chipIdx = long(idx)
+      dataIdx = chipIdx / (1023 * 20)
+      x = message.getBit(dataIdx) * code.getCodeBit(chipIdx)
+      return x
+
+    vdata = scipy.vectorize(dataChip)
+    chips = vdata(chipAll_idx)
+
+    scipy.multiply(signal, chips, signal)
+    return (signal, userTimeX_s, chipAll_idx, chips)
+
+def constDoppler(distance0_m, frequency_hz, doppler_hz):
```

Comment:
What does this wrapper do? Seems like it replicates the functionality of the `Doppler` class.